frontend/src/actions/cartAction.js

```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+import { CART_ADD_ITEM, CART_REMOVE_ITEM } from "../constants/cartConstants.js"
+
+
+export const addToCart = (id, qty) => async (dispatch, getState) => {
+    const { data } = await axios.get(`/api/products/${id}`);
+
+
+    await axios.get(`/api/products`)
+
+    dispatch({
+        type: CART_ADD_ITEM, payload: {
+            product: data._id,
+            name: data.name,
+            image: data.image,
+            price: data.price,
+            countInstock: data.countInstock,
+            qty
+        }
+    })
```

Comment:
Codacy has a fix for the issue: [Replace `··})` with `});`](https://app.codacy.com/gh/Durotemy/e-commerce/pullRequest?prid=9481957)
```suggestion
  });
```